localstack/services/edge.py

```diff
@@ -437,46 +447,52 @@ def start_proxy(port, asynchronous=False):
     return proxy
 
 
-def start_edge(port=None, use_ssl=True, asynchronous=False):
-    if not port:
-        port = config.EDGE_PORT
-    if config.EDGE_PORT_HTTP and config.EDGE_PORT_HTTP != port:
-        do_start_edge(
-            config.EDGE_BIND_HOST,
-            config.EDGE_PORT_HTTP,
-            use_ssl=use_ssl,
-            asynchronous=True,
-        )
-    if port > 1024 or is_root():
-        return do_start_edge(config.EDGE_BIND_HOST, port, use_ssl, asynchronous=asynchronous)
-
-    # process requires privileged port but we're not root -> try running as sudo
-
-    class Terminator:
-        def stop(self, quiet=True):
-            try:
-                url = "http%s://%s:%s" % ("s" if use_ssl else "", LOCALHOST, port)
-                requests.verify_ssl = False
-                requests.post(url, headers={HEADER_KILL_SIGNAL: "kill"})
-            except Exception:
-                pass
-
-    # register a signal handler to terminate the sudo process later on
-    TMP_THREADS.append(Terminator())
-
-    # start the TCP proxy
-    env_vars = {
-        "DEBUG": os.environ.get("DEBUG", ""),
-        "EDGE_FORWARD_URL": config.get_edge_url(),
-        "EDGE_BIND_HOST": config.EDGE_BIND_HOST,
-    }
-    proxy_module = "localstack.services.edge"
-    proxy_args = ["proxy", str(port)]
-    return run_module_as_sudo(
-        module=proxy_module, arguments=proxy_args, env_vars=env_vars, asynchronous=asynchronous
-    )
+def split_list_by(lst: Iterable[T], predicate: Callable[[T], bool]) -> Tuple[List[T], List[T]]:
```

Comment:
nit: This feels like a function which fits well into `localstack.utils.collections`. Maybe move it there?